osimpipeline/vital_tasks.py

```diff
@@ -542,7 +507,327 @@ def __init__(self, trial, ik_setup_task):
+            content = content.replace('@NAME@', self.trial.id)
+            content = content.replace('@DESKINEMATICS@', 
+                self.des_kinematics_fpath)
+
+        with open(target[0], 'w') as f:
+            f.write(content)
+
+    def fill_setup_template(self, file_dep, target, 
+                            init_time=None, final_time=None):
+        with open(file_dep[0]) as ft:
+            content = ft.read()
+            content = content.replace('@STUDYNAME@', self.study.name)
+            content = content.replace('@NAME@', self.trial.id)
+            content = content.replace('@MODEL@', 
+                                      self.adjusted_model_fpath)
+            content = content.replace('@INIT_TIME@', '%.4f' % init_time)
+            content = content.replace('@FINAL_TIME@', '%.4f' % final_time)
+            force_set_files = '%s %s' % (self.cmc_act_fpath,
+                self.res_act_fpath)
+            content = content.replace('@FORCESETFILES@', force_set_files)
+            content = content.replace('@DESKINEMATICS@',
+                self.des_kinematics_fpath)
+            content = content.replace('@CONTROLCONSTRAINTS@', 
+                self.control_constraints_fpath)
+            content = content.replace('@EXTLOADS@', 
+                self.results_extloads_fpath)
+            content = content.replace('@TASKS@', self.results_tasks_fpath)
+        
+        with open(target[0], 'w') as f:
+            f.write(content)
+
+class TaskCMC(task.ToolTask):
+    REGISTRY = []
+    def __init__(self, trial, cmc_setup_task, cycle=None):
+        super(TaskCMC, self).__init__('cmc', trial, cycle=cycle)
+        self.doc = "Run OpenSim's Computed Muscle Control tool."
+        self.des_kinetics_fpath = \
+            '%s/expdata/ground_reaction_orig.mot' % trial.results_exp_path
+
+        if cycle:
+            self.path = os.path.join(trial.results_exp_path, 'cmc', cycle.name)
+        else:
+            self.path = os.path.join(trial.results_exp_path, 'cmc')
+
+        self.file_dep += [
+            cmc_setup_task.adjusted_model_fpath,
+            cmc_setup_task.results_extloads_fpath,
+            cmc_setup_task.results_tasks_fpath,
+            cmc_setup_task.des_kinematics_fpath,
+            self.des_kinetics_fpath,
+            cmc_setup_task.cmc_act_fpath,
+            cmc_setup_task.res_act_fpath,
+            ]
+
+        if cmc_setup_task.control_constraints:
+            self.file_dep += [self.control_contraints_fpath]
+
+        # Set targets for all CMC outputs
+        for cmc_output in ['Actuation_force.sto',
+                'Actuation_power.sto', 'Actuation_speed.sto',
+                'avgResiduals.txt', 'controls.sto', 'controls.xml',
+                'Kinematics_dudt.sto', 'Kinematics_q.sto', 'Kinematics_u.sto',
+                'pErr.sto', 'states.sto']:
+
+            self.targets += ['%s/results/%s_%s_%s_%s' % (
+                self.path, self.study.name, trial.id, 'cmc', cmc_output)]
```

Comment:
This is a good argument for keeping `cycle.id`. Or we could specify a attribute in tool task that covers both cases (either single trial, or trial divided into cycles)